### PR TITLE
Add `Not` verifier for negating verifier results

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/2" target="_blank">Graphite</a> 
      * **PR #3** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/3" target="_blank">Graphite</a> 
        * **PR #9** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/9" target="_blank">Graphite</a> 
          * **PR #10** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/10" target="_blank">Graphite</a> 
            * **PR #11** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/11" target="_blank">Graphite</a> 
              * **PR #12** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/12" target="_blank">Graphite</a> 
              * **PR #18** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/18" target="_blank">Graphite</a> 
                * **PR #20** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/20" target="_blank">Graphite</a> 
                  * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/21" target="_blank">Graphite</a> 
                    * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/22" target="_blank">Graphite</a> 
                      * **PR #23** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/23" target="_blank">Graphite</a> 
                        * **PR #24** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/24" target="_blank">Graphite</a> 
                          * **PR #25** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/25" target="_blank">Graphite</a> 
                            * **PR #26** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/26" target="_blank">Graphite</a> 
                              * **PR #30** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/30" target="_blank">Graphite</a> 
                                * **PR #31** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/31" target="_blank">Graphite</a> 
                                  * **PR #33** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/33" target="_blank">Graphite</a>  👈
    * **PR #4** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/4" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/33?utm_source=stack-comment).

File: verifier/src/lib.rs [hunk 1]
```diff
@@ -564,4 +623,55 @@ mod tests {
...
+                1.into(),
+            )),
+            1.into(),
+        );
+        let displayable = failure.display();
+        let expected = r#"
+            NotError {
+                inner: OrError {
+                    left: Passed,
+                    right: The MiscellaneousSelect did not match expected:MiscellaneousSelect(2) actual:MiscellaneousSelect(3),
```
Comment (nick-mobilecoin) [inline verifier/src/lib.rs:650]: Shown here and in the next test, the displaying of errors doesn't traverse down a "passed" branch.
It seems like the `Not` may be difficult to parse if we choose the current display path.

One thought is to update the `Display` implementation to do something like the below
```console
NotError {
    inner: Passed: OrError {
        left: Passed: (The attributes did not match expected:30  actual:30),
        right: Failed: (The MiscellaneousSelect did not match expected:MiscellaneousSelect(2) actual:MiscellaneousSelect(3)),
    }
}
```
the above example feels hard to parse, especially with the negation of `Not`, and the `Passed` followed by a failure message. 
it could be that the error messages should be updated to be more nuetral, i.e instead of "X did not match" it could say "expected a to match b"

File: verifier/src/lib.rs [hunk 2]
```diff
@@ -303,6 +303,51 @@ impl<T, L: Verifier<T>, R: Verifier<T>> Verifier<T> for Or<L, R> {
...
+impl<E: DisplayableError> DisplayableError for NotError<E> {}
+
+impl<E: DisplayableError> Display for NotError<E> {
+    fn fmt(&self, f: &mut Formatter) -> core::fmt::Result {
+        f.debug_struct("NotError")
+            .field("inner", &format_args!("{:#}", self.inner.display()))
+            .finish()
+    }
+}
+/// Will negate the result of the [`Verifier::verify()`] operation.
```
Comment (samdealy) [inline verifier/src/lib.rs:329]: nit: newline

**Review by samdealy — APPROVED** [review]

File: verifier/src/lib.rs (hunk 2 above)
Comment (nick-mobilecoin) [reply verifier/src/lib.rs:330]: done.

**Review by jcape — APPROVED** [review]

File: verifier/src/lib.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply verifier/src/lib.rs]: This is being discussed on #31

**Comment by nick-mobilecoin** [comment]
bumping @samdealy for re-review due to restack on to main.  #31 still has some work, but it shouldn't hold this up.

**Review by samdealy — APPROVED** [review]
